### DISCUSSION TITLE
webhook to awx prefix

### DISCUSSION THREAD

**Discussion by Dees7** [netbox-community/netbox#6523] [Q&A]
hello.
I have awx with webhook to run deploy.  And I'd like to pass netbox request body to awx. 
AWX requere prefix "extra_vars" in POST like
```{"extra_vars":{"event": "updated", "username": "admin", ... , "tags": []}}```

If "body template" is empty for webhook in Netbox  it send all in body.  But if no "extra_vars" prefix AWX skip vars
```{"event": "updated", "username": "admin", ... , "tags": []}```

I can write template like {"extra_vars":{{data}} .....  prechange  postchange username } but I'd like only to add "extra_vars" before  whole  and every request body.

Is there any variable for a whole body? or Can I add prefix in any ways?

**Comment by Dees7** [comment]
now I hardcoded it
```
diff --git a/netbox/extras/models/models.py b/netbox/extras/models/models.py
index 4917a7e4..74264ba2 100644
--- a/netbox/extras/models/models.py
+++ b/netbox/extras/models/models.py
@@ -151,7 +151,7 @@ class Webhook(models.Model):
         if self.body_template:
             return render_jinja2(self.body_template, context)
         else:
-            return json.dumps(context, cls=JSONEncoder)
+            return '{"extra_vars":' + json.dumps(context, cls=JSONEncoder) + '}'
```

**Reply by candlerb** [reply]
See #6515
